notebooks/COS/SplitTag/SplitTag.ipynb

```diff
@@ -0,0 +1,982 @@
+    "# Get the time data for each exposure\n",
+    "binsize = 10 # Number of seconds in a bin\n",
+    "for i, ctag in enumerate(downloaded_corrtags):\n",
+    "    ax = fig.add_subplot(gs[int(i/2),i%2]) # make the subplot\n",
+    "    ctab = Table.read(ctag,1) # read the data into a table\n",
+    "    event_times = ctab['TIME']\n",
+    "    # plot the histogram of times\n",
+    "    hist = ax.hist(event_times,bins = np.arange(0,max(event_times)+binsize,binsize), alpha = 0.5, label = os.path.basename(ctag),\n",
+    "                   color = 'C0C1C2C3'[2*i:2*i+2])\n",
+    "    if \"zgq\" in ctag: # Add a gray box around the suspected transit\n",
+    "        ax.axvspan(450, 850, color='k', alpha=0.15, label=\"Approximate time of transit\")\n",
+    "    plt.legend(fontsize = 12)\n",
+    "# Format the Figure\n",
+    "plt.suptitle(\"Fig 1.1\\nRough lightcurves of exposures\", fontsize = 20)\n",
+    "fig.text(x = 0.42, y = -0.0001, s = \"Time from exposure start [s]\", \n",
+    "        fontsize = 14)\n",
+    "fig.text(y = 0.42, x = -0.001, rotation = 'vertical', s = f\"Counts in $\\sim${binsize} second bin\", \n",
+    "        fontsize = 14)\n",
+    "plt.tight_layout()\n",
+    "plt.savefig(plots_dir /'compare_exposures.png', bbox_inches = 'tight', dpi = 200)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "driving-illness",
+   "metadata": {},
+   "source": [
+    "We can see an apparent transit in the exposure `lc1va0zgq`, occurring from about 450 - 850 seconds into the exposure. We highlight this in figure 1.2 below.\n",
+    "\n",
+    "Exposure `lc1va0yeq` also seems to contain at least the latter part of a transit, but for now we'll focus on `lc1va0zgq`."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "sized-arizona",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Find the path to the right exposure\n",
+    "transit_exp = [exp for exp in downloaded_corrtags if \"zgq\" in exp][0]\n",
+    "transit_basename = os.path.basename(transit_exp).split('_')[0]\n",
+    "\n",
+    "# Create the plot:\n",
+    "fig = plt.figure(figsize = (12, 6))\n",
+    "\n",
+    "# Grab the data and plot it:\n",
+    "binsize = 5 # Binsize in seconds\n",
+    "ctab = Table.read(transit_exp,1)\n",
+    "event_times = ctab['TIME']\n",
+    "hist = plt.hist(event_times,bins = np.arange(0,max(event_times)+binsize,binsize),\n",
+    "                color = 'k', alpha = 0.7)\n",
+    "\n",
+    "# Add time ranges \"epochs\" or \"windows of time\" as transparent spans:\n",
+    "epoch_markers = [0,550,850, max(event_times)]\n",
+    "print(f\"The length of the exposure is ~{int(max(event_times))} s\")\n",
+    "epoch_labels = [\"Before Transit\", \"During Transit\", \"After Transit\"]\n",
+    "for i in range(3):\n",
+    "    plt.axvspan(epoch_markers[i], epoch_markers[i+1], \n",
+    "                color = 'bgr'[i], alpha = 0.1,\n",
+    "                label = epoch_labels[i])\n",
```

Comment:
There are a number of for-loops throughout the notebook that loop over an index.  This is a matter of personal preference, but I find it cleaner and more pythonic to loop over the lists directly (`for epoch_marker, color in zip(epoch_markers, colors)`) which has the bonus of removing the hardcoded `3`.  This particular case is complicated by the `i+1` offset.  If you wanted to switch to zip instead of the index, you could do that by changing `epoch_markers = [(0, 550), (550, 850), (850, max(event_times))]`, which would then allow for `for epoch_marker, color in zip(epoch_markers, 'bgr')` with `plt.axvspan(epoch_marker[0], epoch_marker[1], ...)`.   If you find looping over the index more readable, then I think I would suggest leaving all cases as-is for self-consistency throughout the notebook.

Reply:
I changed this in all the cases which made sense to me, but please let me know if you think there are any more which still need clearer loops - Thanks!